interface/main/calendar/add_edit_event.php

```diff
@@ -1612,7 +1639,13 @@ function validate(valu) {
     }
     <?php endif; ?>
 
-    return SubmitForm();
+    var submit = submitme(1, undefined, 'theform', collectvalidation);
+
+    if (submit) {
+        $('#form_action').val(valu);
+        SubmitForm();
+    }
+
```

Comment:
Note that editing a recurrent visit is broken:
1. Create a recurrent visit
2. then edit a recurrent visit (for example, set status to something)
3. Then click Save and then click Current
(it should then save the appt as an individual appt and that appt will no longer be a recurrent appt and then close the popup, however instead nothing happens)